fibonacci.js

```diff
@@ -0,0 +1,162 @@
+const loader = document.getElementById("loader");
+const loader2 = document.getElementById("loader2");
+let btn = document.getElementById("btn");
+let result = document.getElementById("result");
+let checked = document.getElementById("flexCheckDefault");
+let dropdown = document.querySelectorAll(".dropdown-item");
+let history = document.getElementById("history");
+
+function removeSpinner() {
+  loader.classList.remove("spinner-border");
+}
+function removeSpinner2() {
+  loader2.classList.remove("spinner-border");
+}
+
+function callServer(number) {
+  fetch(`http://localhost:5050/fibonacci/${number}`)
+    .then((response) => {
+      if (!response.ok) {
+        response.text().then((erText) => {
+          result.innerHTML = "Server Error:" + " " + erText;
+          result.classList.add("text-danger");
+          removeSpinner();
+          removeSpinner2();
+        });
+      } else {
+        return response.json();
+      }
+    })
+    .then((data) => {
+      if (data) {
+        result.innerHTML = data["result"];
+      }
+      removeSpinner();
+      removeSpinner2();
+    });
+  callServerHistory();
+}
+
+btn.addEventListener("click", function () {
+  let nums = parseInt(document.getElementById("number").value);
+  loader.classList.add("spinner-border");
+  loader2.classList.add("spinner-border");
+  let inputField = document.querySelector("input");
+  result.innerText = "";
+  error.innerText = "";
+  error.classList.add("d-none");
+  result.classList.remove("text-danger");
+  let checkbox = document.getElementById("checkbox");
+
+  if (inputField.classList.contains("text-danger", "border-danger")) {
+    inputField.classList.remove("text-danger", "border-danger");
+  }
+
+  if (nums > 50) {
+    error.classList.add("error");
+    error.classList.remove("d-none");
+    error.innerText = "Can't be larger than 50";
+    removeSpinner();
+    removeSpinner2();
+    inputField.classList.add("text-danger", "border-danger");
+    return;
+  }
+  if (checkbox.checked === false) {
+    fibonacci();
+    removeSpinner();
+    removeSpinner2();
+  } else {
+    error.classList.remove("error");
+    callServer(nums);
+    callServerHistory();
+  }
+});
+
+function callServerHistory(sort) {
+  fetch("http://localhost:5050/getFibonacciResults")
+    .then((response) => {
+      return response.json();
+    })
+
+    .then((data) => {
+           
+      if (document.getElementById("historyList") !== null) {
```

Comment:
You should make this html element into a variable at the top (like you did with others) so you don't have to repeat this line and just use the variable.